### PR TITLE
expose: Fix service validation and help message

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

**Comment by kargakis** [comment]
@smarterclayton PTAL

**Comment by mfojtik** [comment]
LGTM

**Comment by mfojtik** [comment]
[merge]

File: pkg/cmd/cli/cmd/expose.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 
 		supportsTCP := false
 		for _, port := range svc.Spec.Ports {
-			if port.Protocol == "TCP" {
+			if strings.ToUpper(string(port.Protocol)) == "TCP" {
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/expose.go]: Why isn't this a constant comparison?

Comment (kargakis) [inline pkg/cmd/cli/cmd/expose.go]: People might have written their own json right?

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/expose.go]: They'll get a validation error - you can't create a port with "tcp"

Comment (kargakis) [inline pkg/cmd/cli/cmd/expose.go]: K, gonna submit a PR reverting this.

**Pull request merged**